core/consensus/grandpa/impl/kusama_hard_forks.hpp

```diff
@@ -71,13 +72,18 @@ namespace kagome::consensus::grandpa {
           "GeYRRPkyi23wSF3cJGjq82117fKJZUbWsAGimUnzb5RPbB1",
           "DzCJ4y5oT611dfKQwbBDVbtCfENTdMCjb4KGMU3Mq6nyUMu",
       };
-      primitives::AuthorityList authorities;
+      AuthorityList authorities;
       crypto::HasherImpl hasher;
       for (auto &id_str : ids_str) {
-        authorities.emplace_back(primitives::Authority{
-            {primitives::decodeSs58(id_str, hasher).value()},
-            1,
-        });
+        auto &authority = authorities.emplace_back();
+
+        auto buff_res = primitives::decodeSs58(id_str, hasher);
+        BOOST_ASSERT(buff_res.has_value());
+        auto &buff = buff_res.value();
+        BOOST_ASSERT(buff.size() == authority.id.size());
+        std::copy(buff.begin(), buff.end(), authority.id.begin());
+
+        authority.weight = 1;
```

Comment:
Remove `std::copy`, use pubkey constructor (may need name if explicit) 
```suggestion
        authorities.emplace_back(primitives::Authority{
            {primitives::decodeSs58(id_str, hasher).value()},
            1,
        });
```

Reply:
`decodeSs58` returns strict type (sr25519) and can not be used for construction ed25519 directly. It is main reason of this change.